storm-core/src/jvm/org/apache/storm/zookeeper/Zookeeper.java

```diff
@@ -0,0 +1,355 @@
+                if (watch) {
+                    return zk.getData().watched().forPath(npath);
+                } else {
+                    return zk.getData().forPath(npath);
+                }
+            }
+        } catch (KeeperException e) {
+            // this is fine b/c we still have a watch from the successful exists call
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+
+        return null;
+    }
+
+    public static Integer getVersion(CuratorFramework zk, String path, boolean watch) throws Exception {
+        String npath = Utils.normalizePath(path);
+        Stat stat = null;
+        if (existsNode(zk, npath, watch)) {
+            if (watch) {
+                stat = zk.checkExists().watched().forPath(Utils.normalizePath(path));
+            } else {
+                stat = zk.checkExists().forPath(Utils.normalizePath(path));
+            }
+            return Integer.valueOf(stat.getVersion());
+        }
+
+        return null;
+    }
+
+    public static List<String> getChildren(CuratorFramework zk, String path, boolean watch) throws RuntimeException {
+
+        try {
+            String npath = Utils.normalizePath(path);
+            if (watch) {
+                return zk.getChildren().watched().forPath(npath);
+            } else {
+                return zk.getChildren().forPath(npath);
+            }
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+    }
+
+    // Deletes the state inside the zookeeper for a key, for which the
+    // contents of the key starts with nimbus host port information
+    public static void deleteDodeBlobstore(CuratorFramework zk, String parentPath, String hostPortInfo) throws RuntimeException {
+        String parentnPath = Utils.normalizePath(parentPath);
+        List<String> childPathList = null;
+        if (existsNode(zk, parentnPath, false)) {
+            childPathList = getChildren(zk, parentnPath, false);
+        }
+        for (String child : childPathList) {
+            if (child.startsWith(hostPortInfo)) {
+                LOG.debug("deleteNode child " + child);
+                deleteNode(zk, parentnPath + "/" + child);
+            }
+        }
+    }
+
+    public static Stat setData(CuratorFramework zk, String path, byte[] data) throws RuntimeException {
+
+        try {
+            String npath = Utils.normalizePath(path);
+            return zk.setData().forPath(npath, data);
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+    }
+
+    public static boolean exists(CuratorFramework zk, String path, boolean watch) throws RuntimeException {
+        return existsNode(zk, path, watch);
+    }
+
+    public static NIOServerCnxnFactory mkInprocessZookeeper(String localdir, Integer port) throws IOException, InterruptedException {
+        LOG.info("Starting inprocess zookeeper at port " + port + " and dir " + localdir);
```

Comment:
`LOG.info("Starting inprocess zookeeper at port {} and dir {}", port, localdir);` instead